src/BookReader.js

```diff
@@ -2227,21 +2227,37 @@ BookReader.prototype._getPageURI = function(index, reduce, rotate) {
 };
 
 /**
- * @param {string}
+ * @param {string} msg
+ * @param {function|undefined} onCloseCallback
  */
-BookReader.prototype.showProgressPopup = function(msg) {
+BookReader.prototype.showProgressPopup = function(msg, onCloseCallback) {
   if (this.popup) return;
 
   this.popup = document.createElement("div");
   $(this.popup).prop('className', 'BRprogresspopup');
-  var bar = document.createElement("div");
+
+  if (typeof(onCloseCallback) === 'function') {
```

Comment:
Is the intention to check for `undefined` or specifically if it's a `function`? If it's the former, I might change the check to that since checking for a function makes me think other things can be passed in.

Reply:
_Suggestion_ I might pull this conditional out to its own method since it's kinda big, ie:
```
if (typeof(onCloseCallback) === 'function') {
  this.addCloseButton(onCloseCallback);
}
```

Reply:
it is chunky, yes. but this file is unique, as it just binds everything to the bookreader object.. i didn't want to muddle those "publicly accessible functions" with this as it is just for that modal...

Reply:
i will hold bc that file is quite brittle 😬  i had to stop myself from making that whole thing a web component 😅